model-engine/model_engine_server/db/models/hosted_model_inference.py

```diff
@@ -453,6 +453,7 @@ class Endpoint(Base):
     current_bundle = relationship("Bundle")
     owner = Column(String(SHORT_STRING))
     public_inference = Column(Boolean, default=False)
+    shadow_endpoints_ids = Column(ARRAY(Text), default=[])
```

Comment:
@song-william guess we're gonna have to figure out an endpoint migration process now heh

Reply:
i don't think shadow needs a migration, but production variants should need a migration

Reply:
@yunfeng-scale why doesn't adding a shadows column require a migration?

Reply:
oh sorry i mean a backfill